### DISCUSSION TITLE
about libfast

### DISCUSSION THREAD

**Discussion by wangyi13683254409** [strongswan/strongswan#1814] [General]
when we use libfast as our web server, can we implement the functionality of download file from explorer.
could you give me a demo implement this function?

**Comment by tobiasbrunner** [comment]
> can we implement the functionality of download file from explorer.

What does that mean?

Please note that _libfast_ is a very old utility library that was created for the deprecated _manager_ web interface and like that is currently not actively maintained.

**Reply by wangyi13683254409** [reply]
Thanks for your reply!
I want to config strongswan through IE, and I want to upload certifications file to server(upload to  /ect/strongswan/swanctl/x509ca/).
I known there is python web server project called strongman. but I want to use C implement the web server.
Do you have any good method to do this?

**Reply by tobiasbrunner** [reply]
> I want to config strongswan through IE

I hope you just mean web browser and not actually "Internet Explorer" as that doesn't exist anymore.

> I want to upload certifications file to server(upload to /ect/strongswan/swanctl/x509ca/)

That sounds risky. And that directory is usually only writable by root, under which you don't want to run your web server.

> I known there is python web server project called strongman.

I'd definitely recommend looking into that instead of implementing something from scratch. strongMan uses [vici](https://docs.strongswan.org/docs/5.9/plugins/vici.html) to configure the daemon, it does not write to `/etc`.

> but I want to use C implement the web server.
> Do you have any good method to do this?

I guess HTTP file uploads may be implemented in C via libfast (or some other similar library). You could have a look at the [medsrv](https://github.com/strongswan/strongswan/tree/master/src/medsrv) web app (not actively maintained either).

**Reply by wangyi13683254409** [reply]
Thanks for your reply in time and thanks for your suggest!

Now we want to config strongswan in web mode and will use get, post and file upload function of web browser.

I read the strongswan document, and got the informations as below.
    manager | A deprecated graphical management application for charon based on libfast
    medsrv   | An experimental management front end for mediation servers based on libfast

I think [manager] is deprecated, so i discard it.
[medsrv] is an experimental management front end, i am afraid of its stability.

anyway, as your suggest, I will research the [medsrv] .  Thanks a lot.
